shared/graphql/schema.graphql

```diff
@@ -860,8 +880,11 @@ type UserAdminMutations {
   mutate(name: String!): MutateUserAdminMutations
 }
 
+union SetSsoConfigResult = Ok | InvalidInput | OperationError
+
 type AdminMutations {
   user: UserAdminMutations!
+  setSsoConfig(config: String): SetSsoConfigResult!
```

Comment:
@sir-sigurd @fiskus i should've mentioned this earlier, but i think we should be returning the new instance of the SsoConfig (with fresh timestamp and such) instead of an empty Ok response -- that way we could update the front-end state without an extra roundtrip

Reply:
https://github.com/quiltdata/enterprise/pull/754

Reply:
So, I update the cache. But cache invalidation still re-request SsoConfig. I don't know how to invalidate everything under `admin` except SsoConfig

Reply:
well, first, you don't have to invalidate everything -- you could invalidate just the relevant things (users collection, default role settings...). second, you should patch the cache to update the result of the ssoConfig query.
maybe it's all not worth it, and invalidating everything is good enough, given it's considerably simpler.

Reply:
ok i see you are updating the `admin.ssoConfig` query (see an inline suggestion on how to fix it).
things to invalidate:
- `roles`
- `defaultRole` (@sir-sigurd actually is this updated automatically?)
- `admin.isDefaultRoleSettingDisabled`
- `admin.user`

Reply:
Yes, but what is the syntax for this? How can I invalidate `admin.isDefaultRoleSettingDisabled`?

Reply:
i think something like this should work:
```ts
cache.invalidate('AdminQueries', 'user')
cache.invalidate('AdminQueries', 'isDefaultRoleSettingDisabled')
```

tho i'm not quite sure, bc AdminQueries doesn't have it's own key afair

the docs: https://commerce.nearform.com/open-source/urql/docs/graphcache/cache-updates/#invalidating-entities

if it's too much work, just keep things simple and invalidate everything

Reply:
Ok, thanks!
I think, it's not worth it, I'm going to invalidate `admin` and `roles`. I think, it's ok for such operation.

Invalidating specific properties kinda leaks the implementation of SSO config, because we need to invalidate individual fields stored in config that only backend knows how to store and parse.

Reply:
> the docs

Is this documentation still relevant? We use v4, but the latest version is v7

Reply:
>  I'm going to invalidate `admin` and `roles`.

So, there is a kinda 3Kb network re-fetching vs, probably, 1 or 2 Kb :)

Reply:
The latest change is:
* cache `SsoConfig` based on timestamp of the `timestamp` property
* invalidate everything under `admin` and `roles` in `Query` after setting new `SsoConfig`

```diff
diff --git a/catalog/app/utils/GraphQL/Provider.tsx b/catalog/app/utils/GraphQL/Provider.tsx
index 75197214..329f6527 100644
--- a/catalog/app/utils/GraphQL/Provider.tsx
+++ b/catalog/app/utils/GraphQL/Provider.tsx
@@ -108,6 +108,8 @@ export default function GraphQLProvider({ children }: React.PropsWithChildren<{}
             p.bucket?.name && p.policy?.id ? `${p.bucket.name}/${p.policy.id}` : null,
           RoleBucketPermission: (p: any) =>
             p.bucket?.name && p.role?.id ? `${p.bucket.name}/${p.role.id}` : null,
+          SsoConfig: (c) =>
+            c.timestamp instanceof Date ? c.timestamp.getTime().toString() : null,
           Status: () => null,
           StatusReport: (r) => (typeof r.timestamp === 'string' ? r.timestamp : null),
           StatusReportList: () => null,
@@ -335,6 +337,10 @@ export default function GraphQLProvider({ children }: React.PropsWithChildren<{}
                   R.evolve({ admin: { user: { list: rmUser } } }),
                 )
               }
+              if (result.admin?.setSsoConfig?.__typename === 'SsoConfig') {
+                cache.invalidate({ __typename: 'Query' }, 'admin')
+                cache.invalidate({ __typename: 'Query' }, 'roles')
+              }
             },
           },
         },
```

Reply:
> > the docs
> 
> Is this documentation still relevant? We use v4, but the latest version is v7

it's mostly ok afaict